src/infosquito/es_crawler.clj

```diff
@@ -81,18 +92,20 @@
 
 (defn- purge-deleted-files
   [es props]
-  (purge-deleted-items es :file icat/file-exists? props))
+  (purge-deleted-items es :file icat/exists? props))
 
 (defn- purge-deleted-folders
   [es props]
   (let [index-base (cfg/get-base-collection props)]
     (purge-deleted-items es
                          :folder
-                         #(and (index/indexable? index-base %) (icat/folder-exists? %))
+                         #(and (index/indexable? index-base %) (icat/exists? %))
                          props)))
 
 (defn purge-index
   [props]
+  (icat/reset-existence-cache)
   (let [es (esr/connect (cfg/get-es-url props))]
     (purge-deleted-files es props)
-    (purge-deleted-folders es props)))
+    (purge-deleted-folders es props))
+  (icat/reset-existence-cache))
```

Comment:
Just being extra careful here, and also trying to make it so this always operates with a clean state and clears unneeded info out of the heap afterwards.